backend/app/controllers/users_controller.rb

```diff
@@ -4,12 +4,39 @@ class UsersController < ApplicationController
   def index
     @users = policy_scope(User).order(:email)
 
-    if users_with_roles_only?
+    users_with_roles_only = params.fetch(:users_with_roles_only, nil)
```

Comment:
you should read params in a homogeneous way. Either use `.fetch` everywhere either get it with:
```suggestion
    users_with_roles_only = params[:users_with_roles_only]
```